src/device/router_device_channels_worker.erl

```diff
@@ -342,40 +343,46 @@ handle_cast(_Msg, State) ->
 %% Channel Handling
 %% ------------------------------------------------------------------
 handle_info(
-    refresh_channels,
+    ?REFRESH_CHANNELS,
     #state{event_mgr = EventMgrRef, device = Device, channels = Channels0} = State
 ) ->
-    APIChannels = lists:foldl(
-        fun(Channel, Acc) ->
-            ID = router_channel:unique_id(Channel),
-            maps:put(ID, Channel, Acc)
-        end,
-        #{},
-        router_console_api:get_channels(Device, self())
-    ),
-    Channels1 =
-        case maps:size(APIChannels) == 0 of
-            true ->
-                %% API returned no channels removing all of them and adding the "no channel"
-                lists:foreach(
-                    fun
-                        ({router_no_channel, <<"no_channel">>}) -> ok;
-                        (Handler) -> gen_event:delete_handler(EventMgrRef, Handler, [])
-                    end,
-                    gen_event:which_handlers(EventMgrRef)
-                ),
-                NoChannel = maybe_start_no_channel(Device, EventMgrRef),
-                #{router_channel:unique_id(NoChannel) => NoChannel};
-            false ->
-                %% Start channels asynchronously
-                lists:foreach(
-                    fun(Channel) -> self() ! {start_channel, Channel} end,
-                    maps:values(APIChannels)
-                ),
-                %% Removing old channels left in cache but not in API call
-                remove_old_channels(EventMgrRef, APIChannels, Channels0)
-        end,
-    {noreply, State#state{channels = Channels1}};
+    case router_console_api:get_channels(Device, self()) of
+        {error, _Reason} ->
+            _ = erlang:send_after(timer:seconds(1), self(), ?REFRESH_CHANNELS),
```

Comment:
We might want to have a retry limit. The other day, a channel was misconfigured on the frontend and wasn't able to be converted, so it would fail, kill the worker, and retry when the next packet came through. 
This would catch that, then retry every second forever.

Reply:
Nevermind, this will only fail if the request fails. 
And malformed channels will continue in the same pattern as before.